### PR TITLE
Tweak the RD's suplexing ability into a skillchip

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: The Research Director's knowledge of suplexing immovable rods is
now in a skillchip.
/:cl:

Although this doesn't change the power level of the Research Director,
it's important to rework "job title" checks into traits or something
similar.

Moving job title specific behaviour out of checks by job name and into
traits means there's more flexibility to trigger behaviour. If an admin
feels the station really needs the ability, they can var edit it in, or
spawn the chip.

### CODE REVIEW

File: code/modules/jobs/job_types/research_director.dm [hunk 1]
```diff
@@ -51,6 +51,8 @@
 	backpack = /obj/item/storage/backpack/science
 	satchel = /obj/item/storage/backpack/satchel/tox
 
+	skillchips = list(/obj/item/skillchip/job/research_director)
```
Comment (ATH1909) [inline code/modules/jobs/job_types/research_director.dm:54]: we really ought to give RDs that roboticist skillchip that tells you which cyborg wire is which

File: code/modules/library/skill_learning/job_skillchips/research_director.dm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/obj/item/skillchip/job/research_director
+	name = "R.D.S.P.L.X skillchip"
```
Comment (Fikou) [inline code/modules/library/skill_learning/job_skillchips/research_director.dm:2]: there should be a dot after X

File: code/modules/jobs/job_types/research_director.dm (hunk 1 above)
Comment (coiax) [reply code/modules/jobs/job_types/research_director.dm:54]: Yes, it would be out of scope. But it's an easy thing to do?

**Comment by JohnFulpWillard** [comment]
Im not a fan of this, I dont like how a chip in your brain would give you superhuman strength capable of suplexing a rod, things like serial numbers and wires make sense because its mind alterations, but strength just doesnt fit that

**Comment by san7890** [comment]
> Im not a fan of this, I dont like how a chip in your brain would give you superhuman strength capable of suplexing a rod, things like serial numbers and wires make sense because its mind alterations, but strength just doesnt fit that

Maybe it's not raw strength that lets an RD take down a rod. Think about it, Research Directors are meant to be dorky geeks in stuffy beige suits and clipboards, yet they can stop an unstoppable rod. It makes more sense for it to be a technical thing that requires a skillset and knowledge, rather than brute force.

File: code/modules/events/immovable_rod.dm [hunk 3]
```diff
@@ -174,22 +174,20 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
...
-				wizard.Stun(60)
-				wizard.apply_damage(25, BRUTE)
-				qdel(src)
-			else
-				U.client.give_award(/datum/award/achievement/misc/feat_of_strength, U) //rod-form wizards would probably make this a lot easier to get so keep it to regular rods only
-				U.visible_message("<span class='boldwarning'>[U] suplexes [src] into the ground!</span>", "<span class='warning'>You suplex [src] into the ground!</span>")
-				new /obj/structure/festivus/anchored(drop_location())
-				new /obj/effect/anomaly/flux(drop_location())
-				qdel(src)
+	if(HAS_TRAIT(user, TRAIT_ROD_SUPLEX) || (user.mind && HAS_TRAIT(user.mind, TRAIT_ROD_SUPLEX)))
```
Comment (Rohesie) [inline code/modules/events/immovable_rod.dm:177]:
```suggestion
	if(!HAS_TRAIT(user, TRAIT_ROD_SUPLEX) && !(user.mind && HAS_TRAIT(user.mind, TRAIT_ROD_SUPLEX)))
		return
```
Could you reduce this proc's horizontal drift with early returns?

File: code/modules/library/skill_learning/job_skillchips/research_director.dm [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/obj/item/skillchip/job/research_director
+	name = "R.D.S.P.L.X. skillchip"
+	desc = "Knowledge of how to solve the ancient conumdrum; what happens when an unstoppable force meets an immovable object."
+	auto_trait = TRAIT_ROD_SUPLEX
+	skill_name = "True Strength"
+	skill_description = "The knowledge and strength to resolve the most ancient conumdrum; what happens when an unstoppable force meets an immovable object."
+	skill_icon = "dumbbell"
+	activate_message = "<span class='notice'>You feel the urge to prove yourself in the upcoming Feats of Strength.</span>"
+	deactivate_message = "<span class='notice'>You no care about the upcoming Feats of Strength, and are no longer clear on what they were in the first place.</span>"
```
Comment (Rohesie) [inline code/modules/library/skill_learning/job_skillchips/research_director.dm:9]: Could you change this message to be a bit nerdier?
It's the Research Director, not an athlete.

File: code/modules/events/immovable_rod.dm [hunk 5]
```diff
@@ -174,22 +174,28 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
...
+		wizard.apply_damage(25, BRUTE)
+		qdel(src)
+	else
+		user.client.give_award(/datum/award/achievement/misc/feat_of_strength, user) //rod-form wizards would probably make this a lot easier to get so keep it to regular rods only
+		user.visible_message("<span class='boldwarning'>[user] suplexes [src] into the ground!</span>", "<span class='warning'>You suplex [src] into the ground!</span>")
+		new /obj/structure/festivus/anchored(drop_location())
+		new /obj/effect/anomaly/flux(drop_location())
+		qdel(src)
+
+	. = TRUE
```
Comment (Rohesie) [inline code/modules/events/immovable_rod.dm:201]:
```suggestion
	return TRUE
```

File: code/modules/events/immovable_rod.dm [hunk 6]
```diff
@@ -174,22 +174,28 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
...
+	if(.)
+		return
+
+	if(!(HAS_TRAIT(user, TRAIT_ROD_SUPLEX) || (user.mind && HAS_TRAIT(user.mind, TRAIT_ROD_SUPLEX))))
+		return
+
+	playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+	for(var/mob/M in urange(8, src))
+		if(!M.stat)
+			shake_camera(M, 2, 3)
```
Comment (Rohesie) [inline code/modules/events/immovable_rod.dm:187]:
```suggestion
		if(M.stat)
			continue
		shake_camera(M, 2, 3)
```

**Pull request merged**